R/dataframe-frame.R

```diff
@@ -1888,3 +1888,103 @@ dataframe__write_parquet <- function(
     invisible(self)
   })
 }
+
+#' Unstack a long table to a wide form without doing an aggregation
+#'
+#' This can be much faster than a pivot, because it can skip the grouping phase.
+#'
+#' @inheritParams rlang::args_dots_empty
+#' @param step Number of rows in the unstacked frame.
+#' @param how Direction of the unstack. Must be one of `"vertical"` or
+#' `"horizontal"`.
+#' @param columns Column name(s) or selector(s) to include in the operation. If
+#' set `NULL` (default), use all columns.
+#' @param fill_values Fill values that don’t fit the new size with this value.
+#'
+#' @inherit as_polars_df return
+#' @examples
+#' df <- pl$DataFrame(x = LETTERS[1:8], y = 1:8)$with_columns(
+#'   z = pl$int_ranges(pl$col("y"), pl$col("y") + 2, dtype = pl$UInt8)
+#' )
+#' df
+#'
+#' df$unstack(step = 4, how = "vertical")
+#' df$unstack(step = 2, how = "horizontal")
+#' df$unstack(step = 5, columns = cs$numeric(), fill_values = 0)
+dataframe__unstack <- function(
+  ...,
+  step,
+  how = c("vertical", "horizontal"),
+  columns = NULL,
+  fill_values = NULL
+) {
+  wrap({
+    check_dots_empty0(...)
+
+    how <- arg_match0(how, values = c("vertical", "horizontal"))
+    if (length(columns) > 0) {
+      if (is_polars_selector(columns)) {
+        df <- self$select(columns)
+      } else {
+        df <- self$select(!!!columns)
+      }
+    } else {
+      df <- self
+    }
```

Comment:
I don't remember if we support selectors in dynamic dots or not. If so, `columns` can be replaced by those